### ISSUE TITLE
How can I set custom optimizer scheduler?

### ISSUE THREAD

**Issue by YuNie24** [NVIDIA/apex#1617]
I'm using APEX like below,

`
optimizer = torch.optim.Adam(model.parameters(), 
                            lr=opt.lr, betas=(opt.beta1, 0.999), 
                            weight_decay=opt.weight_decay)

model, optimizer = amp.initialize(model, optimizer, opt_level='O0')
`

However, I want to use warm up learning rate scheduler and so on. 
How can I apply custom scheduler on AMP optimizer??

If I don't use APEX, I can do like below,

`
class NoamOpt:
    def __init__(self, model_size, factor, warmup, optimizer, init_step=0):
        self.optimizer = optimizer
        self._step = init_step
        self.warmup = warmup
        self.factor = factor
        self.model_size = model_size
        self._rate = 0
        
    def step(self):
        "Update parameters and rate"
        self._step += 1
        rate = self.rate()
        for p in self.optimizer.param_groups:
            p['lr'] = rate
        self._rate = rate
        self.optimizer.step()
        
    def rate(self, step = None):
        "Implement `lrate` above"
        if step is None:
            step = self._step
        return self.factor * \
            (self.model_size ** (-0.5) *
            min(step ** (-0.5), step * self.warmup ** (-1.5)))
    
optimizer = torch.optim.Adam(model.parameters(),  lr=opt.lr, betas=(opt.beta1, 0.999),  weight_decay=opt.weight_decay)
optimizer = NoamOpt(opt.trn_d_model, 1, 2000, optimizer,  init_step=start_iter)
`
